File: src/main/scala/jk_5/nailed/server/map/game/script/api/ScriptObjectiveApi.scala
package jk_5.nailed.server.map.game.script.api

import jk_5.nailed.api.scoreboard.Objective

/**
 * No description given
 *
 * @author jk-5
 */
class ScriptObjectiveApi(private[this] val objective: Objective) {

  def setDisplayName(name: String) = objective.setDisplayName(name)
  def set(score: String, value: Int) = objective.getScore(score).setValue(value)
  def add(score: String, value: Int) = objective.getScore(score).addValue(value)
  def get(score: String) = objective.getScore(score).getValue

  def getObjectiveId = objective.getId
}

File: src/main/java/jk_5/nailed/server/map/game/script/api/ScriptObjectiveApi.java
package jk_5.nailed.server.map.game.script.api;

import jk_5.nailed.api.scoreboard.Objective;

public class ScriptObjectiveApi {

    private final Objective objective;

    public ScriptObjectiveApi(Objective objective) {
        this.objective = objective;
    }

    public void setDisplayName(String name){
        objective.setDisplayName(name);
    }

    public void set(String score, int value){
        objective.getScore(score).setValue(value);
    }

    public void add(String score, int amount){
        objective.getScore(score).addValue(amount);
    }

    public int get(String score){
        return objective.getScore(score).getValue();
    }

    public String getObjectiveId(){
        return objective.getId();
    }
}
